From 42ba34f93aa1ee2082505ac1055b7a366fb59c17 Mon Sep 17 00:00:00 2001
From: Leon Schumacher <leonsch@protonmail.com>
Date: Mon, 17 Feb 2025 17:15:41 +0100
Subject: [PATCH] fix help links

---
 src/libcmd/markdown.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libcmd/markdown.cc b/src/libcmd/markdown.cc
index dbaab8c19..394c07d8b 100644
--- a/src/libcmd/markdown.cc
+++ b/src/libcmd/markdown.cc
@@ -47,7 +47,7 @@ std::string renderMarkdownToTerminal(std::string_view markdown)
     if (!rndr_res)
         throw Error("allocation error while rendering Markdown");
 
-    return filterANSIEscapes(std::string(buf->data, buf->size), !shouldANSI());
+    return std::string(buf->data, buf->size);
 }
 
 }
-- 
2.47.2

