From 2807d6bc04baefe0d309337ddd70f8b2cfa56eb7 Mon Sep 17 00:00:00 2001
From: Leon Schumacher <leonsch@protonmail.com>
Date: Sun, 29 Dec 2024 19:47:27 +0100
Subject: [PATCH] arg_int.c: include ctype.h

---
 src/arg_int.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/arg_int.c b/src/arg_int.c
index 29c20e5..f805244 100644
--- a/src/arg_int.c
+++ b/src/arg_int.c
@@ -26,6 +26,7 @@ USA.
 
 /* #ifdef HAVE_STDLIB_H */
 #include <stdlib.h>
+#include <ctype.h>
 /* #endif */
 
 #include "argtable2.h"
-- 
2.47.0

