diff --git a/scripts/NR.filter b/scripts/NR.filter
index 3ad4341..2f86947 100755
--- a/scripts/NR.filter
+++ b/scripts/NR.filter
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/bin/sh
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
 grep __NR_ | sed 's/#define //g' | cut -d ' ' -f 1
diff --git a/scripts/NR.generate b/scripts/NR.generate
index 063eb76..980d8f3 100755
--- a/scripts/NR.generate
+++ b/scripts/NR.generate
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/bin/sh
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
 while read -r NR; do
diff --git a/scripts/test.bash b/scripts/test.bash
index 9c3b787..8b26c12 100755
--- a/scripts/test.bash
+++ b/scripts/test.bash
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/bin/sh
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
 test_suite="${1}"
diff --git a/scripts/test.new b/scripts/test.new
index 66be525..afa8419 100755
--- a/scripts/test.new
+++ b/scripts/test.new
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/bin/sh
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
 if [[ -z "${1}" ]]; then
