diff --git a/src/screens/media_library.cpp b/src/screens/media_library.cpp
index 558458d..f873704 100644
--- a/src/screens/media_library.cpp
+++ b/src/screens/media_library.cpp
@@ -52,8 +52,8 @@ MediaLibrary *myLibrary;
 
 namespace {
 
-bool hasTwoColumns;
-bool isAlbumOnly;
+bool hasTwoColumns = 1;
+bool isAlbumOnly = 1;
 size_t itsLeftColStartX;
 size_t itsLeftColWidth;
 size_t itsMiddleColWidth;
@@ -189,8 +189,8 @@ MediaLibrary::MediaLibrary()
 , m_window_timeout(Config.data_fetching_delay ? 250 : BaseScreen::defaultWindowTimeout)
 , m_fetching_delay(boost::posix_time::milliseconds(Config.data_fetching_delay ? 250 : -1))
 {
-	hasTwoColumns = 0;
-	isAlbumOnly = 0;
+	hasTwoColumns = 1;
+	isAlbumOnly = 1;
 	itsLeftColWidth = COLS/3-1;
 	itsMiddleColWidth = COLS/3;
 	itsMiddleColStartX = itsLeftColWidth+1;
@@ -232,6 +232,8 @@ MediaLibrary::MediaLibrary()
 	));
 	
 	w = &Tags;
+
+	toggleColumnsMode();
 }
 
 void MediaLibrary::resize()
@@ -904,15 +906,8 @@ void MediaLibrary::updateTimer()
 
 void MediaLibrary::toggleColumnsMode()
 {
-	if (isAlbumOnly)
-	{
-		hasTwoColumns = 0;
-		isAlbumOnly = 0;
-	}
-	else if (hasTwoColumns)
-		isAlbumOnly = 1;
-	else
-		hasTwoColumns = 1;
+	hasTwoColumns = 1;
+	isAlbumOnly = 1;
 
 	Tags.clear();
 	Albums.clear();
